test/e2e/network/netpol/kubemanager.go

```diff
@@ -179,6 +179,7 @@ func (k *kubeManager) probeConnectivity(args *probeConnectivityArgs) (bool, stri
 
 	commandDebugString := fmt.Sprintf("kubectl exec %s -c %s -n %s -- %s", args.podFrom, args.containerFrom, args.nsFrom, strings.Join(cmd, " "))
 
+	maxAttempts := 10
```

Comment:
This probably needs to be time-based, not retry-count-based. The problem here (IIRC) is that there's no indication of when the network plugin is done reprogramming rules after a pod/namespace/policy change, so we can't do the equivalent of `WaitForEndpointOnNode` or something like that; we have to just poll until the policy seems to be working, or we give up.

Reply:
I changed that in the last commit, this logic was also causing a lot of problems because it was execing each time, that is very expensive as it has to create a tunnel over the apiserver

Reply:
the assertion is tricky indeed